src/core/public/capabilities/capabilities_service.test.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { InjectedMetadataService } from '../injected_metadata';
+import { CapabilitiesService } from './capabilities_service';
+
+describe('#start', () => {
+  it('returns a service with getCapabilities', () => {
+    const injectedMetadata = new InjectedMetadataService({
+      injectedMetadata: {
+        vars: {
+          uiCapabilities: {
+            foo: 'bar',
+            bar: 'baz',
+          },
+        },
+      } as any,
+    });
+    const service = new CapabilitiesService();
+    const startContract = service.setup({ injectedMetadata: injectedMetadata.setup() });
+    expect(startContract.getCapabilities()).toEqual({
+      foo: 'bar',
+      bar: 'baz',
+    });
+  });
+
+  it(`does not allow Capabilities to be modified`, () => {
+    const injectedMetadata = new InjectedMetadataService({
+      injectedMetadata: {
+        vars: {
+          uiCapabilities: {
+            foo: 'bar',
+            bar: 'baz',
+          },
+        },
+      } as any,
+    });
+    const service = new CapabilitiesService();
+    const startContract = service.setup({ injectedMetadata: injectedMetadata.setup() });
+    const capabilities = startContract.getCapabilities();
+
+    // @ts-ignore TypeScript knows this shouldn't be possible
+    expect(() => (capabilities.foo = 'foo')).toThrowErrorMatchingInlineSnapshot(
```

Comment:
can we just test assigning throws? matching on TS error message leads to additional work to update tests when they change formats